Lab_3

sergeiev22 opened the pull request:
В задании 8 не смог построить решение на базе функции for. Не понимаю логику построения решения.

sergeiev22 committed:
Lab_3

EgorOb commented on Lab_3/Task_8.py:
```diff
@@ -0,0 +1,17 @@
+
+for i in range(capital):
+    month = capital - spend
+    spend = spend * 1.05
```
1.05 тут как магическая константа, а она зависит от increase
`spend *= 1 + increase`
На базе for не рационально строить, так как не знаем заранее сколько циклов сделаем.
Тут лучше через while делать
```
while money_capital - spend > 0:
    money_capital += salary - spend:
    spend *= 1 + increase
    month += 1
```

EgorOb commented on Lab_3/Task_9.py:
```diff
@@ -0,0 +1,16 @@
+
+for i in range(months):
+    need_money = need_money + spend
+    spend = spend * 1.03
```
Опять магическая 1.03

EgorOb reviewed (changes requested):
8 - ую задачу можно переделать по коду который я представил

sergeiev22 committed:
Update Task_9.py

Исправил. Убрал магическое число, заменил на строчку: spend *= 1 + increase

sergeiev22 committed:
Update Task_8.py

Избавился в решении от магического числа. Заменил решение на цилке for на цикл while

EgorOb reviewed (approved):
Зачтено
